spec/standard-library.md

```diff
@@ -1,730 +1,535 @@
+
+> [!NOTE]
+> **This text is informative.**
+>
+> The standard library is intended to be the minimum set of types and members required by a conforming C# implementation. As such, it contains only those members that are explicitly required by the C# language specification.
+>
+>It is expected that a conforming C# implementation will supply a significantly more extensive library that enables useful programs to be written. For example, a conforming implementation might extend this library by
+>
+> - Adding namespaces.
+> - Adding types.
+> - Adding members to non-interface types.
+> - Adding intervening base classes or interfaces.
+> - Having struct and class types implement additional interfaces.
+> - Adding attributes (other than the ConditionalAttribute) to existing types and members.
+>
+> **End of informative text.**
 
-## General {#general-108 .Appendix2}
+## Standard Library Types defined in ISO/IEC 23271
 
-A conforming C\# implementation shall provide a minimum set of types having specific semantics. These types and their members are listed here, in alphabetical order by namespace and type. For a formal definition of these types and their members, refer to ISO/IEC 23271:2012 *Common Language Infrastructure (CLI), Partition IV; Base Class Library (BCL), Extended Numerics Library, and Extended Array Library*, which are included by reference in this specification.
+```csharp
 
-**This text is informative.**
-
-The standard library is intended to be the minimum set of types and members required by a conforming C\# implementation. As such, it contains only those members that are explicitly required by the C\# language specification.
-
-It is expected that a conforming C\# implementation will supply a significantly more extensive library that enables useful programs to be written. For example, a conforming implementation might extend this library by
-
--   Adding namespaces.
-
--   Adding types.
-
--   Adding members to non-interface types.
-
--   Adding intervening base classes or interfaces.
-
--   Having struct and class types implement additional interfaces.
-
--   Adding attributes (other than the ConditionalAttribute) to existing types and members.
-
-**End of informative text.**
-
-## Standard Library Types defined in ISO/IEC 23271 {#standard-library-types-defined-in-isoiec-23271 .Appendix2}
-
-namespace System\
-{\
-public class ArgumentException : SystemException\
-{\
-public ArgumentException();\
-public ArgumentException(string message);\
-public ArgumentException(string message, Exception innerException);\
-}\
+namespace System {
```

Comment:
In [corefx/coding-style.md at master · dotnet/corefx](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md ) NO.1, we use Allman style braces, where each brace begins on a new line. A single line statement block can go without braces but the block must be properly indented on its own line and must not be nested in other statement blocks that use braces (See issue 381 for examples). One exception is that a using statement is permitted to be nested within another using statement by starting on the following line at the same indentation level, even if the nested using contains a controlled block."